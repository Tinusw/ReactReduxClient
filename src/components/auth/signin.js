import React, { Component } from "react";
import { connect } from "react-redux";
import * as actions from "../../actions";

export class SignIn extends Component {
  shouldComponentUpdate(nextProps) {
    if (nextProps.authenticated === true) {
      this.props.history.push("/campaign/index");
    }
    return true;
  }

  constructor() {
    super();
    this.state = {
      email: "",
      password: ""
    };
  }

  handleInputChange = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };

  handleSubmit = event => {
    event.preventDefault();
    this.props.signinUser(this.state);
  };

  render() {
    return (
      <form className="SignIn" onSubmit={this.handleSubmit}>
        <label>email</label>
        <input
          id="email"
          onChange={this.handleInputChange}
          name="email"
          type="text"
        />
        <label>password</label>
        <input
          id="password"
          onChange={this.handleInputChange}
          name="password"
          type="text"
        />
        <button>Submit</button>
      </form>
    );
  }
}

function mapStateToProps(state) {
  return {
    errorMessage: state.auth.error,
    authenticated: state.auth.authenticated
  };
}

export default connect(mapStateToProps, actions)(SignIn);
