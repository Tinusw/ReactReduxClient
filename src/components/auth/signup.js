import React, { Component } from "react";
import { connect } from "react-redux";
import * as actions from "../../actions";

export class SignUp extends Component {
  constructor() {
    super();
    this.state = {
      email: "",
      password: "",
      passwordConfirm: ""
    };
  }

  shouldComponentUpdate(nextProps) {
    if (nextProps.authenticated === true) {
      this.props.history.push("/campaign/index");
    }
    return true;
  }

  handleInputChange = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };

  handleSubmit = event => {
    event.preventDefault();
    this.props.signUpUser(this.state);
  };

  renderError() {
    if (
      this.state.passwordConfirm !== "" &&
      this.state.password !== "" &&
      this.state.password !== this.state.passwordConfirm
    ) {
      return (
        <div className="error" id="error">
          passwords do not match :(
        </div>
      );
    }
  }

  render() {
    return (
      <form className="SignUp" onSubmit={this.handleSubmit}>
        <label>email</label>
        <input
          id="email"
          onChange={this.handleInputChange}
          name="email"
          type="text"
          autoComplete="email"
        />
        <label>password</label>
        <input
          id="password"
          onChange={this.handleInputChange}
          name="password"
          type="password"
        />
        <label>confirm password</label>
        <input
          id="passwordConfirm"
          onChange={this.handleInputChange}
          name="passwordConfirm"
          type="password"
        />
        <button>Sign Up</button>
        {this.renderError()}
      </form>
    );
  }
}

function mapStateToProps(state) {
  return {
    errorMessage: state.auth.error,
    authenticated: state.auth.authenticated
  };
}

export default connect(mapStateToProps, actions)(SignUp);
