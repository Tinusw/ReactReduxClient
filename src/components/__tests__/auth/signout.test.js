import React from "react";
import * as test_tools from "../../../test/test_helper.js";

import { SignOut } from "../../../components/auth/signout";

describe("<SignOut/>", () => {
  const mockSignOutfn = jest.fn();

  let actual;

  beforeEach(() => {
    actual = shallow(<SignOut signoutUser={mockSignOutfn} />);
  });

  it("renders correctly", () => {
    expect(enzymeToJson(actual)).toMatchSnapshot();
  });

  it("should have the className Signout", () => {
    expect(actual.find("div.Signout").length).toBe(1);
    expect(actual.find("div.Signout").props().children).toEqual("Signed Out");
  });

  it("should call signoutUser on mount at least once", () => {
    expect(mockSignOutfn.mock.calls.length).toBeGreaterThan(0);
  });
});
