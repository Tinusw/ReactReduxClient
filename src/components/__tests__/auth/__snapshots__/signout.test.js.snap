// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignOut/> renders correctly 1`] = `
<div
  className="Signout"
>
  Signed Out
</div>
`;
